fe/fe-core/src/main/java/org/apache/doris/nereids/trees/expressions/ComparisonPredicate.java

```diff
@@ -58,4 +59,16 @@ public AbstractDataType inputType() {
      * Commute between left and right children.
      */
     public abstract ComparisonPredicate commute();
+
+    @Override
+    public TypeCheckResult checkInputDataTypes() {
+        for (int i = 0; i < super.children().size(); i++) {
+            Expression input = super.children().get(i);
+            if (input.getDataType().isBitmapType()) {
```

Comment:
use `isObjectType()` to ban all object types

Reply:
done